graphics/macgui/macfontmanager.cpp

```diff
@@ -640,6 +645,65 @@ int MacFontManager::registerFontName(Common::String name, int preferredId) {
 	return id;
 }
 
+int MacFontManager::registerTTFFont(TTFMap *ttfList) {
+	int defaultValue = kMacFontGeneva;
+	int realId = 100;
+	auto checkId = [&](int id) {
+		int i = 0;
+		while (ttfList[i].ttfName != "0") {
```

Comment:
Where does this magic value `"0"` come from?

Reply:
I thought it was you who asked me to change the `ttfFamily[]` into this form before, so I used "0" to determine the end condition.

Reply:
> sev 2024/02/23 21:23
> yes, I did not pay much attention
> please change it to
> ```cpp
> const Graphics::TTFMap *ttfFamily[] = {
>     { .... }, 
>     {0, 0} };
> ```
> e.g. it is a static array with end marker

Reply:
There is a major difference between '0', which is then could be used as an end marker, and '"0"', which is a string.

Reply:
done the changes